doc/tutorials/03_Interactivity.ipynb

```diff
@@ -0,0 +1,311 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "05e2ed65-fa51-4613-9cc1-4aae1843f275",
+   "metadata": {},
+   "source": [
+    "<style>div.container { width: 100% }</style>\n",
+    "<img style=\"float:left;  vertical-align:text-bottom;\" height=\"65\" width=\"172\" src=\"../_static/logo_stacked.png\" />\n",
+    "<div style=\"float:right; vertical-align:text-bottom;\"><h2>Tutorial 3. Interactivity</h2></div>\n",
+    "<br><br>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2b059ba6-2c56-498b-a2d1-5532cb64011a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import panel as pn\n",
+    "\n",
+    "pn.extension('tabulator')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "34d5b9b1-33e1-4070-b327-e69c229044c9",
+   "metadata": {},
+   "source": [
+    "## **Interactivity**: Effectively using APIs for adding interactivity and reactivity\n",
+    "\n",
+    "In the previous section we learned a lot about Param, which underlies not only the core architecture of Panel itself but also provides the foundation to adding interactivity to your applications. In this section we will discover how to leverage Parameters and dependencies on parameters to add interactivity. In particular we will focus on implementing interactivity through reactivity, rather than the more imperative style of programming you might be used to from other UI frameworks.\n",
+    "\n",
+    "By the end of this tutorial you should have learned:\n",
+    "\n",
+    "- How to use both declarative and imperative APIs to achieve interactivity\n",
+    "- How to develop both functional and class based interactive apps"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f4ab18fa-3f81-43aa-a4de-bc10f6da3f8e",
+   "metadata": {},
+   "source": [
+    "## Declarative vs. Imperative programming\n",
+    "\n",
+    "To build an interactive component in Panel we have two options, either we define callbacks that perform explicit actions, e.g. \"when this widget changes update this component\", or we declare a reactive function or method that returns a specific output given some input.\n",
+    "\n",
+    "Let's look at what this looks like in practice by building a simple app that allows us to select a subset of columns to display in a table.\n",
+    "\n",
+    "### Imperative\n",
+    "\n",
+    "In both cases we start by loading our data and then defining a widget that will allow us to interact with the data:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b0cedaaa-c66d-4ea9-83de-09b8a1e4bc6c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "turbines = pd.read_parquet('./windturbines.parq')\n",
```

Comment:
This tutorial doesn't work for me because I don't have windturbines.parq

Maybe you could choose example data for the tutorials that can be readily loaded (e.g. https://docs.bokeh.org/en/latest/docs/examples/basic/data/transform_markers.html)?

Or alternatively link to an online URL or bundle it so that the notebook executes?

Reply:
I'm probably either going to bundle it or load it from our cdn.